### PR TITLE
Added plugin.name to fish tag log lines

### PR DESCRIPTION
**Pull request by andsel**
Added the plugin's name to log lines generated by the plugins

----

This PR adds `plugin.id` to log lines.

Tipical use case:
- the user sees some log lines with `plugin.id` bring that id (running **Java pipeline**)
- query the LS Node Info API to get source reference in config file for that id (`curl -XGET "localhost:9600/_node/pipelines/main?graph=true"`)

**How to test**
- run logstash with java pipeline, with debug log running the following pipeline.conf:
```
input {
  stdin {}
}

filter {
  json {
    source => "message"
  }
  sleep {
    id => "sleep_filter_123"
    time => 1
  }
}

output {
  stdout {
    codec => rubydebug
  }
}
```
`bin/logstash -f "<config>" --java-execution=true  --debug`
- in console or in `logs/logstash-plain.log` you should find a line like:
```
[DEBUG][logstash.filters.json    ][main][31543d02478e75487017793958c743af64669fe5f915ab05695717240d026f49] Running json filter {:event=>#<LogStash::Event:0x2220404c>}
```
- bring that uuid and search in Node Into API with a rest call

### CODE REVIEW

**Comment by andsel** [comment]
Ok I do it!

**Comment by andsel** [comment]
@robbavey this is done and ready to review

File: qa/integration/specs/plugin_name_log_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  let(:config) { @fixture.config("root") }
+
+  it "should write logs with plugin name" do
+    settings = {
+      "path.logs" => temp_dir,
+      "log.level" => "debug"
+    }
+    IO.write(@ls.application_settings_file, settings.to_yaml)
+    @ls.spawn_logstash("-w", "1" , "-e", config)
+    #@ls.wait_for_logstash
```
Comment (robbavey) [inline qa/integration/specs/plugin_name_log_spec.rb]: Nit: Please remove comments

Comment (andsel) [reply qa/integration/specs/plugin_name_log_spec.rb]: Fixed, thanks @robbavey.
I've also create an issue #11154 and a PR #11155 to start discussing a useful `plugin.id` or `plugin.name`

File: logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import org.jruby.internal.runtime.methods.DynamicMethod;
 import org.jruby.runtime.Block;
 import org.jruby.runtime.builtin.IRubyObject;
+import org.jruby.runtime.ThreadContext;
```
Comment (robbavey) [inline logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java]: Nit: unused import

File: logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputDelegatorExt.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import org.jruby.runtime.Block;
 import org.jruby.runtime.ThreadContext;
 import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.RubyUtil;
```
Comment (robbavey) [inline logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputDelegatorExt.java]: Nit: Unused import

File: logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java (hunk 2 above)
Comment (andsel) [reply logstash-core/src/main/java/org/logstash/config/ir/compiler/DatasetCompiler.java]: Done

File: logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputDelegatorExt.java (hunk 3 above)
Comment (andsel) [reply logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputDelegatorExt.java]: Done

**Pull request closed without merging**